src/swarm-scenario/Swarm/Game/Scenario.hs

```diff
@@ -355,10 +357,15 @@ instance FromJSONE ScenarioInputs Scenario where
           namedGrids = map stuffGrid mergedStructures
           recognizableGrids = filter Structure.isRecognizable namedGrids
 
-      symmetryAnnotatedGrids <- mapM checkSymmetry recognizableGrids
+          -- We exclude empty grids from the recognition engine.
+          nonEmptyRecognizableGrids = mapMaybe (traverse getNonEmptyGrid) recognizableGrids
+
+      myAutomatons <-
+        either (fail . T.unpack . renderRedundancy) return $
+          mkAutomatons (fmap cellToEntity) nonEmptyRecognizableGrids
```

Comment:
@byorgey here is one interesting change.  Previously the orientations to be recognized were checked for redundancy (according to symmetry) in the scenario parser.  I have now moved that detail to within the recognizer API.

Related to this, we now instantiate the recognizer engine from within the scenario parsing code (instead of from within game state initialization), so that it can throw an error when a redundant recognition spec is provided.